graphene_django/static/graphene_django/graphiql.js

```diff
@@ -61,17 +61,29 @@
   var fetchURL = locationQuery(otherParams);
 
   // Defines a GraphQL fetcher using the fetch API.
-  function httpClient(graphQLParams) {
-    var headers = {
-      Accept: "application/json",
-      "Content-Type": "application/json",
-    };
+  function httpClient(graphQLParams, opts = { headers: {} }) {
+    let headers = opts.headers;
+    // Convert headers to an object.
+    if (typeof headers === 'string') {
+      headers = JSON.parse(opts.headers);
+    }
     if (csrftoken) {
-      headers["X-CSRFToken"] = csrftoken;
+      Object.assign(
+          {
+            'X-CSRFToken': csrftoken
+          },
+          headers,
+      )
     }
     return fetch(fetchURL, {
       method: "post",
-      headers: headers,
+      headers: Object.assign(
+          {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+          },
+          headers,
+      ),
```

Comment:
Similar issue with es6+. You can avoid it by mutating headers object earlier in code:
```js
headers['Accept'] = headers['Accept'] || 'application/json';
headers['Content-Type'] = headers['Content-Type'] || 'application/json';
```